src/main/java/org/whispersystems/textsecuregcm/storage/Accounts.java

```diff
@@ -84,6 +88,10 @@
   @SqlQuery("SELECT * FROM accounts WHERE " + NUMBER + " > :from ORDER BY " + NUMBER + " LIMIT :limit")
   public abstract List<Account> getAllFrom(@Bind("from") String from, @Bind("limit") int length);
 
+  @Mapper(ActiveUserMapper.class)
+  @SqlQuery("SELECT " + ID + " as id, (devices->>'lastSeen') as lastSeen, (devices->>'id') as deviceId, case when (devices->>'gcmId') is not null then " + PLATFORM_ID_ANDROID + " when (devices->>'apnId') is not null then " + PLATFORM_ID_IOS + " else " + PLATFORM_ID_UNKNOWN + " end as platformId FROM accounts a, json_array_elements(a.data->'devices') devices WHERE " + ID + " > :from ORDER BY " + ID + " LIMIT :limit")
+  public abstract List<ActiveUser> getActiveUsersFrom(@Bind("from") long from, @Bind("limit") int length);
```

Comment:
I think the basic rationale comes from efficiency. 
Sending the entire account record (several hundred bytes) over the wire to the frontend servers felt wasteful compared to the 20 bytes required to actually do the work. I was also concerned that if I wanted to use a larger chunk size (to increase throughput) that it would cause a lot more memory usage than necessary.

Reply:
now using the KV model in e432b60ae2e03314836d6cdf270fe75ed286a9ca